src/core/border_router/routing_manager.hpp

```diff
@@ -648,22 +685,33 @@ class RoutingManager : public InstanceLocator
         // interface prefix, maintaining the discovered prefix
         // lifetime, and selection of the NAT64 prefix to publish in
         // Network Data.
+        //
+        // Calling methods except GeneratrLocalPrefix and SetEnabled
+        // when disabled becomes no-op.
 
         explicit Nat64PrefixManager(Instance &aInstance);
 
-        void               Start(void);
-        void               Stop(void);
+        void OnRoutingManagerStart(void);
+        void OnRoutingManagerStop(void);
```

Comment:
The caller may may expect `Start` / `Stop` to unconditionally start / stop the prefix manager.

One other consideration is `StartIfEnabled()`, but I want to have a symmetric name as `Stop`, since the `Stop` is unconditional.

Another consideration is `Error Start()`, which will return `kErrorInvalidState` if it is disabled. But I also want to have the same function signature as other prefix managers, so the Start / Stop should not have return value.

I think `OnRoutingManagerStart` fits the need that hide some details of Nat64PrefixManager from RoutingManager, and the NAT64 prefix manager will take care of the state by it self.